pkg/dns/ibm/public/cis_provider.go

```diff
@@ -160,41 +166,45 @@ func (p *Provider) createOrUpdateDNSRecord(record *iov1.DNSRecord, zone configv1
 	// "." when it creates a wildcard DNS record.
 	dnsName := strings.TrimSuffix(record.Spec.DNSName, ".")
 	listOpt.SetName(dnsName)
-	for _, target := range record.Spec.Targets {
-		listOpt.SetContent(target)
```

Comment:
Wow, how did anything work with this here?  But do we want to add just one target to the listOpt, or is that the issue you're fixing - we don't want to match on name + target?